### ISSUE TITLE
Indentation: possible false errors after a real one

### ISSUE THREAD

**Issue by pbludov** [checkstyle/checkstyle#4593]
$ javac TestClass.java

$ cat TestClass.java

```
package test;

public class TestClass {

        @Override       // [OK] 'method def modifier' has incorrect indentation level 8, expected level should be 4.
    public String
        toString() {    // [Why?] 'toString' has incorrect indentation level 8, expected level should be 12.

        return null;
    }

        @Override       // [OK] 'method def modifier' has incorrect indentation level 8, expected level should be 4.
    public boolean equals(
        Object b) {     // [Why?] 'Object' has incorrect indentation level 8, expected level should be 12.

        return false;
    }
}

```

$ cat TestConfig.xml

```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
  "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
  <property name="charset" value="UTF-8"/>
  <module name="TreeWalker">
    <module name="Indentation"/>
    </module>
  </module>
</module>

```

````
$ java -jar checkstyle-X.XX-all.jar -c config.xml YOUR_FILE.java

k$ java -jar checkstyle-8.0-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:5: 'method def modifier' has incorrect indentation level 8, expected level should be 4. [Indentation]
[ERROR] TestClass.java:7: 'toString' has incorrect indentation level 8, expected level should be 12. [Indentation]
[ERROR] TestClass.java:12: 'method def modifier' has incorrect indentation level 8, expected level should be 4. [Indentation]
[ERROR] TestClass.java:14: 'Object' has incorrect indentation level 8, expected level should be 12. [Indentation]
Audit done.
Checkstyle ends with 4 errors.
````

There should be only two indentation errors.
Bad indentation on an annotation implies a false bad indentation on method name or parameter.

It is not a real bug: correction of the indentation of the annotation also "corrects" the next (false) error.

**Comment by romani** [comment]
I doubt we can do anything there. As first line of structure/block/statement is incorrect, all child/following are calculated base on its indentation.
But there is some inconsistency in behavior
```
$ cat TestClass.java 
public class TestClass {
        @Override
    public String
        toString() {
        return null;
        }
}

$ java -jar checkstyle-7.8.2-all.jar -c config.xml TestClass.java
Starting audit...
[ERROR] /var/tmp/TestClass.java:2: 'method def modifier' has incorrect indentation level 8, expected level should be 4. [Indentation]
[ERROR] /var/tmp/TestClass.java:4: 'toString' has incorrect indentation level 8, expected level should be 12. [Indentation]
[ERROR] /var/tmp/TestClass.java:6: 'method def rcurly' has incorrect indentation level 8, expected level should be 4. [Indentation]
Audit done.
Checkstyle ends with 3 errors.
```

Attention to all 3 violations, 1st is expected, second if based on actual wrong indentation on first, third is independent as 1st.

Looks like it is related to AST structure 
```
$ java -jar checkstyle-7.8.2-all.jar -t TestClass.java
CLASS_DEF -> CLASS_DEF [1:0]
|--MODIFIERS -> MODIFIERS [1:0]
|   `--LITERAL_PUBLIC -> public [1:0]
|--LITERAL_CLASS -> class [1:7]
|--IDENT -> TestClass [1:13]
`--OBJBLOCK -> OBJBLOCK [1:23]
    |--LCURLY -> { [1:23]
    |--METHOD_DEF -> METHOD_DEF [2:8]
    |   |--MODIFIERS -> MODIFIERS [2:8]
    |   |   |--ANNOTATION -> ANNOTATION [2:8]
    |   |   |   |--AT -> @ [2:8]
    |   |   |   `--IDENT -> Override [2:9]
    |   |   `--LITERAL_PUBLIC -> public [3:4]
    |   |--TYPE -> TYPE [3:11]
    |   |   `--IDENT -> String [3:11]
    |   |--IDENT -> toString [4:8]
    |   |--LPAREN -> ( [4:16]
    |   |--PARAMETERS -> PARAMETERS [4:17]
    |   |--RPAREN -> ) [4:17]
    |   `--SLIST -> { [4:19]
    |       |--LITERAL_RETURN -> return [5:8]
    |       |   |--EXPR -> EXPR [5:15]
    |       |   |   `--LITERAL_NULL -> null [5:15]
    |       |   `--SEMI -> ; [5:19]
    |       `--RCURLY -> } [6:8]
    `--RCURLY -> } [7:0]
```

@rnveach , please give us insight on what is expected.

**Comment by rnveach** [comment]
> Looks like it is related to AST structure

Almost all issues with Indentation are related to differences between file layout and tree structure.

> please give us insight on what is expected.

This has been a gray area for me. Some places will base it on parent's position, others will force it a specific place.
One example is class definition of file must be declared at position 0, any other position isn't allowed.

I personally think we should we should be sending user to correct indentation always, so maybe we should change it to force all to correct position and disregard parent's location and go by what parent's location should be.

**Comment by pbludov** [comment]
If this can't be easily fixed, then there is no reason to waste time on it.
